### PR TITLE
Use dotenv-safe

### PR DESCRIPTION
**Pull request by nwinch**
This branch replaces the combination of `dotenv` + `required_env` with `dotenv-safe`.

`dotenv` is great and while I was reading the docs I noticed there exists a `dotenv-safe` which can be used in place of `dotenv` to do what `required_env` does.

I think it makes sense to deal with environment variables in one single fashion.  The advantages I see are:
- One file for environment vars (`.env`) and one file for defaults (`.env.default`)
- Both files sit right next to each other, which is a more logical positioning.
- Both `.env` and `.env.default` are the same format.
- Cleaner and less code for initialization.

### CODE REVIEW

**Comment by davidbanham** [comment]
That is super cool! Haven't seen that package before. The only concern I have is that in order to run the migrations you need to define _all_ environment variables whereas now it can just pull in only the `db.json` file. Maybe that's not actually worth caring about, though. Maybe it's actually a negative feature since what if someone writes a migration that expects some other env var to be set? The migration tests wouldn't actually pick that up since the tests load all env vars.

Yeah, I think this is a good thing.

File: base/dotenv.default.mustache [hunk 1]
```diff
@@ -0,0 +1,3 @@
+PORT=
+RETHINK_HOST=
+RETHINK_NAME=
```
Comment (davidbanham) [inline base/dotenv.default.mustache]: Why are these separate from dotenv.mustache ?

Comment (nwinch) [inline base/dotenv.default.mustache]: This file is used as the template to check which `env` vars you should have listed.

Comment (davidbanham) [inline base/dotenv.default.mustache]: Oh, I see. Why not just use env.example as the template though?

Comment (nwinch) [inline base/dotenv.default.mustache]: I didn't use it because it seemed to exist for the purpose of simply using as a template to create a `.env`, so I was concerned that users might remove `example.env` after creating a `.env`, and not realising it was used for checking env vars exist.

But I like the idea as it's one less file.

Actually, perhaps we do use it, and it's also a free way to provide users a notification to create an `.env` file?

... actually actually, why do we create `example.env` and not just `.env`?

Comment (davidbanham) [inline base/dotenv.default.mustache]: We use example.env rather than .env because .env should never evar evar be committed to source control. The danger is that you put some testing specific stuff in .env. You accidentally commit that change to git. That file then wends its way through to staging and/or prod with the testing keys/config still in there and Bad Things Happen.

So, we create example.env to protect against that eventuality, but still make it simple for people to just `cp example.env .env` and get on with writing some code.

I don't think we need to worry about people deleting example.env since that would be a silly thing to do. It would mean that any future travellers to the project would have no idea what was expected to be in the .env file.

tl:dr, let's ditch dotenv.default and just use the example one.

Comment (nwinch) [inline base/dotenv.default.mustache]: .env should be in .gitingore so it won't be committed?  So I think creating a .env would be okay?

Hmm I'm still concerned with the naming of "example".. just sounds like turfable file/code.

Comment (davidbanham) [inline base/dotenv.default.mustache]: It is `.gitignore`d, but I like to take an onion approach with these things. The more tools we can use to stop testing data ending up in prod the better.

It sounds like we've reached an accord on the fact that there's no need for three files. We're now shooting for whether we can actually take two files down to one, correct? I think that is a reasonable question to ask, and one to which the answer may well be "Yeah, actually that would be good!", but we should not let it hold us up moving to dotenv-safe. If we're happy, let's just get this using the example then pull the trigger on this PR.

Comment (nwinch) [inline base/dotenv.default.mustache]: Yep, sounds good to me :)

I'll update using the example file.

**Comment by davidbanham** [comment]
@nwinch I've implemented that change we talked about using the example file and removing the dotenv default file. Can you please review?

**Comment by nwinch** [comment]
LGTM 👍
